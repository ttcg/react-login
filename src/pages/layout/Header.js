import React from 'react';
import { connect } from 'react-redux'
import {
    Nav,
    Navbar
} from 'react-bootstrap';
import { NavLink, Link } from 'react-router-dom';
import {
    logoutUser
} from '../../redux/security/security.action'

export const Header = ({ 
    email, 
    isLoggedIn,
    onLogOut }) => {
    return (
        <Navbar bg="dark" variant="dark">
            <Link className="navbar-brand" to="/">React-Login
            </Link>
            <Navbar.Toggle aria-controls="basic-navbar-nav" />
            <Navbar.Collapse id="basic-navbar-nav">
                <Nav className="mr-auto">
                    <NavLinkWrapper to="/" exact>Home</NavLinkWrapper>
                </Nav>
            </Navbar.Collapse>
            {
                isLoggedIn &&
                <Navbar.Text>
                    <span style={{ color: 'white' }}>{email}</span> {' '}
                    <button className="btn btn-sm btn-outline-light my-2 my-sm-0" onClick={onLogOut}>Logout</button>
                </Navbar.Text>
            }
        </Navbar>
    );
}

const NavLinkWrapper = ({ to, children, exact }) => (
    <NavLink
        className="nav-link"
        activeClassName="active"
        to={to}
        exact={exact}>
        {children}
    </NavLink>
)

const mapStateToProps = (state) => {
    return {
        email: state.Security.email,
        isLoggedIn: state.Security.isLoggedIn
    };
};

const mapDispatchToProps = {
    onLogOut: logoutUser
}

export default connect(mapStateToProps, mapDispatchToProps)(Header);