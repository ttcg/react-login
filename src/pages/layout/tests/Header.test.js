import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Header } from '../Header';

it('should render Header', () => {
    const email = "test@yopmail.com"
    const component = shallow(<Header
        isLoggedIn={true}
        email={email} />);

    expect(component.text()).toContain(email);
    expect(toJson(component)).toMatchSnapshot();
});