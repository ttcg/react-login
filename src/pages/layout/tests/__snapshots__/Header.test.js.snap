// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header 1`] = `
<Bootstrap(undefined)
  bg="dark"
  variant="dark"
>
  <Link
    className="navbar-brand"
    to="/"
  >
    React-Login
  </Link>
  <NavbarToggle
    aria-controls="basic-navbar-nav"
    label="Toggle navigation"
  />
  <NavbarCollapse
    id="basic-navbar-nav"
  >
    <Nav
      className="mr-auto"
      fill={false}
      justify={false}
    >
      <NavLinkWrapper
        exact={true}
        to="/"
      >
        Home
      </NavLinkWrapper>
    </Nav>
  </NavbarCollapse>
  <NavbarText>
    <span
      style={
        Object {
          "color": "white",
        }
      }
    >
      test@yopmail.com
    </span>
     
     
    <button
      className="btn btn-sm btn-outline-light my-2 my-sm-0"
    >
      Logout
    </button>
  </NavbarText>
</Bootstrap(undefined)>
`;
